src/core/qgsfileutils.cpp

```diff
@@ -56,6 +56,38 @@ QStringList QgsFileUtils::extensionsFromFilter( const QString &filter )
   return extensions;
 }
 
+bool QgsFileUtils::fileMatchesFilter( const QString &fileName, const QString &filter )
```

Comment:
wondeering how this behaves for filters or filenames without extension (e.g. bash scripts or README)

Reply:
I did originally have a test in place for this, but had to remove it as the fallback I've used for qt < 5.12 doesn't support it. Given we'll be forcing 5.12+ in the near future I'll resurrect this test at that time